engines/native/sandbox.go

```diff
@@ -60,8 +77,8 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 	}
 
 	env["HOME"] = homeFolder.Path()
```

Comment:
Maybe instead of calling this variable for `homeFolder, err := b.engine.environment.TemporaryStorage.NewFolder()` we should call it for `workFolder`, still set it as working directory, and use it for homeDir if we're creating a new user...